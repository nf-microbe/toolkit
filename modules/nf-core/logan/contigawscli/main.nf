process LOGAN_CONTIGAWSCLI {
    tag "${meta.id}"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/seqkit_awscli_biopython:4bbd3cad980280ad' :
        'community.wave.seqera.io/library/seqkit_awscli_biopython:1ebd36d34b5de691' }"

    input:
    tuple val(meta), val(accession)

    output:
    tuple val(meta), path("${accession}.contigs.fa.gz") , emit: fasta
    path "versions.yml"                                 , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args    = task.ext.args ?: ''
    prefix  = task.ext.prefix ?: "${meta.id}"
    """
    ### Download Logan contigs
    aws s3 cp s3://logan-pub/c/${accession}/${accession}.contigs.fa.zst . --no-sign-request || true

    zstd -d ${accession}.contigs.fa.zst

    gzip ${accession}.contigs.fa
    rm -rf ${accession}.contigs.fa.zst

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        awscli: \$( aws --version | sed 's/aws-cli\\///; s/ Python.*//' )
        seqkit: \$(seqkit version | cut -d' ' -f2)
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo "" | gzip > ${accession}.contigs.fa.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        awscli: \$( aws --version | sed 's/aws-cli\\///; s/ Python.*//' )
        seqkit: \$(seqkit version | cut -d' ' -f2)
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
    END_VERSIONS
    """
}
