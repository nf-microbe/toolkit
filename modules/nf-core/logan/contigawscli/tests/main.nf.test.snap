{
    "SRR24085752": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRR24085752",
                            "single_end": false,
                            "md5_1": "e67296355980997dde170c602d9cb4c1",
                            "md5_2": "86b4ee5d4860aed0a3a2538a7757cbc8"
                        },
                        "SRR24085752.contigs.fa.gz:md5,5673115466baa83cc844215658ef356f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,aa8d131d33f1ce17df9609c2286e9692"
                ],
                "fasta": [
                    [
                        {
                            "id": "SRR24085752",
                            "single_end": false,
                            "md5_1": "e67296355980997dde170c602d9cb4c1",
                            "md5_2": "86b4ee5d4860aed0a3a2538a7757cbc8"
                        },
                        "SRR24085752.contigs.fa.gz:md5,5673115466baa83cc844215658ef356f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aa8d131d33f1ce17df9609c2286e9692"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-26T14:10:46.668697572"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRR24085752",
                            "single_end": false,
                            "md5_1": "e67296355980997dde170c602d9cb4c1",
                            "md5_2": "86b4ee5d4860aed0a3a2538a7757cbc8"
                        },
                        "SRR24085752.contigs.fa.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "1": [
                    "versions.yml:md5,aa8d131d33f1ce17df9609c2286e9692"
                ],
                "fasta": [
                    [
                        {
                            "id": "SRR24085752",
                            "single_end": false,
                            "md5_1": "e67296355980997dde170c602d9cb4c1",
                            "md5_2": "86b4ee5d4860aed0a3a2538a7757cbc8"
                        },
                        "SRR24085752.contigs.fa.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,aa8d131d33f1ce17df9609c2286e9692"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-26T14:11:06.328122085"
    }
}